src/pages/Login.tsx

```diff
@@ -5,30 +5,115 @@ import { handleLogin } from '../handlers/authHandler';
 const Login: React.FC = observer(() => {
   const [email, setEmail] = useState('');
   const [password, setPassword] = useState('');
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<string | null>();
 
   const onSubmit = async (event: React.FormEvent) => {
     event.preventDefault();
+    setLoading(true);
+    setError(null);
     try {
       await handleLogin({ email, password });
       // Redirect or update UI on successful login
     } catch (error) {
-      // Show error message
+      setError('Invalid email or password');
+    } finally {
+      setLoading(false);
     }
   };
 
   return (
-    <form onSubmit={onSubmit}>
-      <div>
-        <label>Email</label>
-        <input type="email" value={email} onChange={(e) => setEmail(e.target.value)} required />
-      </div>
-      <div>
-        <label>Password</label>
-        <input type="password" value={password} onChange={(e) => setPassword(e.target.value)} required />
-      </div>
-      <button type="submit">Login</button>
-    </form>
+    <div style={styles.container}>
+      <form onSubmit={onSubmit} style={styles.form}>
+        <h2 style={styles.heading}>Login</h2>
+        <p style={styles.errorMsg}>{error && (error)}</p>
+        <div style={styles.formGroup}>
+          <label htmlFor="email" style={styles.label}>
+            Email
+          </label>
+          <input
+            type="email"
+            id="email"
+            value={email}
+            onChange={(e) => setEmail(e.target.value)}
+            style={styles.input}
+            required
+          />
+        </div>
+        <div style={styles.formGroup}>
+          <label htmlFor="password" style={styles.label}>
+            Password
+          </label>
+          <input
+            type="password"
+            id="password"
+            value={password}
+            onChange={(e) => setPassword(e.target.value)}
+            style={styles.input}
+            required
+          />
+        </div>
+        <button type="submit" style={styles.submitButton} disabled={loading}>
+          {loading ? 'Loading...' : 'Login'}
+        </button>
+      </form>
+    </div>
   );
 });
 
+const styles = {
+  container: {
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+    height: 'calc(100vh - 50px)',
```

Comment:
should be calculated in the root dom, then the rest is flex

Reply:
Done